client/blocks/conversations/list.jsx

```diff
@@ -166,6 +166,47 @@ export class ConversationCommentList extends React.Component {
 		return { ...startingExpanded, ...expansions };
 	};
 
+	setActiveReplyComment = commentId => {
+		const siteId = get( this.props, 'post.site_ID' );
+		const postId = get( this.props, 'post.ID' );
+		this.props.setActiveReply( {
+			siteId,
+			postId,
+			commentId,
+		} );
+	};
+
+	resetActiveReplyComment = () => {
+		this.setActiveReplyComment( null );
+	};
+
+	renderCommentForm = () => {
+		const post = this.props.post;
+		const commentText = this.state.commentText;
+
+		// Are we displaying the comment form at the top-level?
+		if (
+			this.props.activeReplyCommentId ||
+			size(
```

Comment:
@samouri suggested an improvement to this:

https://github.com/Automattic/wp-calypso/pull/18356#discussion_r142994387

Reply:
echoing what I said there, you likely want `some` here. it'll bail early if it finds something and doesn't allocate a new trash array.

Reply:
👍 switched to use `some`, but not `commentsTree.children` as it doesn't seem to include placeholders.